odk2-src/build-app.rst

```diff
@@ -0,0 +1,778 @@
+
+If there are other problems, the contents of the JavaScript Console will be helpful to the ODK core team for debugging. Open the JavaScript Console by clicking the icon with the three bars in the top right, select :guilabel:`More Tools`, select :guilabel:`Developer Tools`, and then select the :guilabel:`Console` tab. Select all of the debugging output, then copy it, save it to a file, and post it to the |forum|_ or create a ticket on the `Github Issue Tracker <https://github.com/opendatakit/opendatakit/issues>`_.
+
+.. _build-app-move-to-device:
+
+Moving Files To The Device
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+  You must have USB debugging enabled on your device in order to perform this step. See `these instructions <https://www.phonearena.com/news/How-to-enable-USB-debugging-on-Android_id53909>`_ for help.
+
+In order to see these changes on an Android device, you must first have ODK Survey installed on your device. Then:
+
+  #. Connect the device to your computer via a USB cable
+  #. Open a :program:`cmd` or :program:`terminal` window within the :guilabel:`Application Designer` directory (the one containing :file:`Gruntfile.js`), as described in the :doc:`app-designer-directories` documentation.
+  #. Type:
+
+.. code-block:: console
+
+  $ grunt adbpush
+
+.. note::
+
+  If it gives you an error, you may need to run :code:`grunt adbpush -f` to force it.
+
+This will copy all of the files under config onto your device. You should then be able to launch ODK Survey, and it will display your form in its list of forms. Click the form to open it.
+
+More :program:`grunt` commands can be found in :ref:`build-app-pushing`.
+
+
+
+.. _build-app-design-view:
+
+ODK Tables: Designing a Custom View
+-------------------------------------
+
+One of the most powerful aspects of ODK Tables is its ability to run HTML and
+JavaScript pages as the skin of the app. Through a JavaScript API presented to these files, you can query the database and control the app.
+
+Writing an app using HTML and JavaScript yields a lot of power. However, it can lead to a complicated design cycle.
+
+The HTML and JavaScript files you write rely on the JavaScript API implemented within the ODK Tables APK to vend data-table values into your HTML pages, where they can be displayed as a list of items, as a detail view of a single item, or graphed in any number of ways. This JavaScript API, since it is implemented in the APK, makes it difficult to debug your custom views off the phone. At present, the only way to test your HTML pages is on the device. Fortunately, on Android 4.4 and higher, :program:`Chrome` can access the browser Console and set breakpoints on the device, providing a clumsy but viable debug environment.
```

Comment:
It may be clearer to change this to "The HTML and JavaScript files you write rely on the JavaScript API implemented within the ODK Tables APK to retrieve database values for your application."  I think the messaging is good, but the wording could be more concise.